react/components/ProgressBar/utils.ts

```diff
@@ -1,31 +1,33 @@
 import { FIFTH_STEP, THIRD_STEP } from './constants'
 
 export function generateProgressBarStates(
-  states: any[],
+  progressBarStates: any[],
   currentState: number,
   packages: any
 ) {
-  return states.map((state, index) => {
-    let label = null
-    if (
-      index === FIFTH_STEP &&
-      currentState === FIFTH_STEP &&
-      isDelivered(packages)
-    ) {
-      label = state.done
-    } else if (
-      index === FIFTH_STEP &&
-      currentState === FIFTH_STEP &&
-      isOrderReadyToPickUp(packages)
-    ) {
-      label = state.doing_pickup
-    } else if (currentState === index) {
-      label = state.doing
-    } else if (currentState > index) {
+  const wasDelivered = isDelivered(packages)
+  const isPickup = isOrderPickUp(packages)
+
+  return progressBarStates.map((_state, index) => {
+    let label
+    const state = _state.pickup
+      ? isPickup
+        ? _state.pickup
+        : _state.shipping
+      : _state
+
+    if (index < currentState) {
       label = state.done
+    } else if (index == currentState) {
+      if (index == FIFTH_STEP && wasDelivered) {
```

Comment:
Any special reasons for using shallow equality in these lines?

Reply:
Nice catch! Fixing it